sbi/neural_nets/mdn.py

```diff
@@ -43,6 +45,13 @@ def build_mdn(
     """
     x_numel = batch_x[0].numel()
     # Infer the output dimensionality of the embedding_net by making a forward pass.
+    assert batch_x.device == batch_y.device, (
+        "Mismatch in fed data's device: "
+        f"batch_x has device '{batch_x.device}' whereas "
+        f"batch_x has device '{batch_x.device}'. Please "
```

Comment:
typo: `y` vs `x`

Reply:
given we have >5 of those asserts you could write a function that checks two tensors for their device and prints a (slightly more general) error message?

Reply:
Will do !